ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -392,107 +343,55 @@ private void createBanner() {
       return;
     }
 
-    cleanUp();
+    VungleManager.getInstance().cleanUpActiveBannerAd(mPlacementId);
     RelativeLayout.LayoutParams adParams =
         new RelativeLayout.LayoutParams(
             RelativeLayout.LayoutParams.WRAP_CONTENT, RelativeLayout.LayoutParams.WRAP_CONTENT);
     adParams.addRule(RelativeLayout.CENTER_HORIZONTAL, RelativeLayout.TRUE);
     adParams.addRule(RelativeLayout.CENTER_VERTICAL, RelativeLayout.TRUE);
-    VungleListener listener = mVungleListener.get();
+    MediationBannerListener listener = bannerListener.get();
 
     if (AdConfig.AdSize.isBannerAdSize(mAdConfig.getAdSize())) {
       mVungleBannerAd = Banners.getBanner(mPlacementId, mAdConfig.getAdSize(), mAdPlayCallback);
       if (mVungleBannerAd != null) {
         Log.d(TAG, "display banner:" + mVungleBannerAd.hashCode() + this);
-        VungleManager.getInstance().storeActiveBannerAd(mPlacementId, this);
+        VungleManager.getInstance().storeActiveBannerAd(mPlacementId, this, true);
         updateVisibility(mVisibility);
         mVungleBannerAd.setLayoutParams(adParams);
         // don't add to parent here
-        if (listener != null) {
-          listener.onAdAvailable();
+        if (listener != null && adapter.get() != null) {
+          listener.onAdLoaded(adapter.get());
         }
       } else {
         // missing resources
-        if (listener != null) {
-          listener.onAdFailedToLoad(AdRequest.ERROR_CODE_INTERNAL_ERROR);
+        if (listener != null && adapter.get() != null) {
+          listener.onAdFailedToLoad(adapter.get(), AdRequest.ERROR_CODE_INTERNAL_ERROR);
         }
       }
     } else {
       View adView = null;
       mVungleNativeAd = Vungle.getNativeAd(mPlacementId, mAdConfig, mAdPlayCallback);
       if (mVungleNativeAd != null) {
         adView = mVungleNativeAd.renderNativeView();
-        VungleManager.getInstance().storeActiveBannerAd(mPlacementId, this);
+        VungleManager.getInstance().storeActiveBannerAd(mPlacementId, this, true);
       }
       if (adView != null) {
         Log.d(TAG, "display MREC:" + mVungleNativeAd.hashCode() + this);
         updateVisibility(mVisibility);
         adView.setLayoutParams(adParams);
         // don't add to parent here
-        if (listener != null) {
-          listener.onAdAvailable();
+        if (listener != null && adapter.get() != null) {
+          listener.onAdLoaded(adapter.get());
         }
       } else {
         // missing resources
-        if (listener != null) {
-          listener.onAdFailedToLoad(AdRequest.ERROR_CODE_INTERNAL_ERROR);
+        if (listener != null && adapter.get() != null) {
+          listener.onAdFailedToLoad(adapter.get(), AdRequest.ERROR_CODE_INTERNAL_ERROR);
         }
       }
     }
   }
 
-  private VungleListener mVungleBannerListener =
```

Comment:
there was some business logic here, better to keep it and we don't need if (listener != null)  checks above
just checks for adapter.get inside

Reply:
done.